### DISCUSSION TITLE
[react] defaultValue in HTMLAttributes seems too narrow?

### DISCUSSION THREAD

**Discussion by marcomuser** [DefinitelyTyped/DefinitelyTyped#71452] [Issues with a @types package]
The [defaultValue](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/cebb88fecfa52f854826e216a537867c11b0150e/types/react/index.d.ts#L2677) attribute in the HTMLAttributes interface is specified as `string | number | readonly string[] | undefined;`. Is there a specific reason why this union is specified this narrowly? I can also pass `bigint | readonly (string|number|bigint)[]` and possibly other types (i.e. `Date`, `boolean`), without causing react warnings/errors or other malfunctions in the browser, while the defaultValue is correctly set to the html element.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
React will always stringify anyway so anything is possible. The current type is just a concesion to existing code. Just pass `defaultValue={String(value)}` to make it explicit.
